psyneulink/core/compositions/composition.py

```diff
@@ -6598,14 +6642,17 @@ def run(
             # Simulations are run as part of the controller node wrapper.
             assert not is_simulation
             try:
+                comp_ex_tags = frozenset({})
+                # Add learning tags if the composition is to be ran in learning mode
+                if self._is_learning(context):
+                    comp_ex_tags = comp_ex_tags.union(frozenset({"learning"}))
```

Comment:
no need to create multiple frozensets and unions:
`comp_ex_tags = frozenset(("learning")) if self._is_learning(context) else frozenset()`
or even
`comp_ex_tags = frozenset(("learning") if self._is_learning(context) else ())`